docs: translate useState page (#535)
reactjs/id.react.dev: src/content/reference/react/useState.md

<Intro>

> `useState` merupakan React Hook yang memungkinkan Anda untuk menambahkan [variabel *state*](/learn/state-a-components-memory) pada komponen anda.

Reviewer: "Anda", always use "A" capital.

Reviewer 2: [Reference](https://ejaan.kemdikbud.go.id/eyd/penggunaan-huruf/huruf-kapital/#23-huruf-kapital-digunakan-sebagai-huruf-pertama-kata-penunjuk-hubungan-kekerabatan-seperti-bapak-ibu-kakak-dan-adik-serta-kata-atau-ungkapan-lain-termasuk-unsur-bentuk-ulang-utuh-yang-digunakan-sebagai-sapaan):

![image](https://user-images.githubusercontent.com/6315466/236875185-8d26d367-1042-4f80-b5b6-836b9df305f2.png)

Author: Thank you for the reference!

  // ...
```

Konvensi dalam menamai variable *state* adalah menggunakan pola `[something, setSomething]` dengan [*array destructuring*.](https://javascript.info/destructuring-assignment)

> [Lihat contoh lainnya di bawah ini.](#usage)

Reviewer: ```suggestion
[Lihat lebih banyak contoh di bawah ini.](#usage)
```

#### Parameter {/*parameters*/}

* `initialState`: Nilai awal pada sebuah *state*. Nilainya dapat berupa jenis apa saja, namun terdapat perilaku khusus untuk fungsi. Argumen ini diabaikan setelah *rendering* awal.
  * Jika Anda mengoper sebuah fungsi sebagai `initialState`, itu akan diperlakukan sebagai fungsi inisialisasi (*initializer function*). Fungsi tersebut harus murni (*pure*), tidak boleh menerima argumen, dan harus mengembalikan nilai dengan tipe apa pun. React akan memanggil fungsi inisialisasi ketika menginisialisasi komponen, dan menyimpan nilai kembaliannya sebagai *state* awal. [Lihat contoh lainnya di bawah ini.](#avoiding-recreating-the-initial-state)

#### Kembalian {/*returns*/}

`useState` mengembalikan sebuah senarai yang memiliki dua nilai:

> 1. *State* saat ini. Pada saat pertama kali *render*, itu akan sama dengan `initialState` yang anda oper sebelumnya.

Reviewer: Anda, with A capital

> 1. *State* saat ini. Pada saat pertama kali *render*, itu akan sama dengan `initialState` yang anda oper sebelumnya.

Reviewer: Anda, with A capital

* Memanggil fungsi `set` selama *rendering* hanya diperbolehkan dari dalam komponen yang sedang di-*render* saat ini. React akan membuang hasilnya dan segera mencoba me-*render* kembali dengan *state* yang baru. Pola ini jarang dibutuhkan, tetapi bisa digunakan untuk **menyimpan informasi dari render sebelumnya**. [See an example below.](#storing-information-from-previous-renders)

* Pada mode ketat, React akan **memanggil fungsi pembaruan Anda dua kali** untuk [membantu Anda menemukan kejadian yang tidak diharapkan.](#my-initializer-or-updater-function-runs-twice) Ini hanya terjadi di lingkungan pengembangan dan tidak memengaruhi produksi. Jika fungsi pembaruan Anda murni (seperti seharusnya), ini tidak akan memengaruhi perilakunya. Hasil dari salah satu panggilan akan diabaikan.

---

## Penggunaan {/*usage*/}

> ### Menambahkan *state* pada sebuah komponen {/*adding-state-to-a-component*/}

Reviewer 3: ```suggestion
### Menambahkan state pada sebuah komponen {/*adding-state-to-a-component*/}
```

Don't italicise loanwords in headings.

https://github.com/reactjs/id.react.dev/wiki/Panduan-Penulisan-Umum#kata-serapan-di-dalam-heading

<Solution />

> #### *Text field* (*string*) {/*text-field-string*/}

Reviewer 3: ```suggestion
#### Text field (string) {/*text-field-string*/}
```

<Solution />

> #### *Checkbox* (*boolean*) {/*checkbox-boolean*/}

Reviewer 3: ```suggestion
#### Checkbox (boolean) {/*checkbox-boolean*/}
```

<Solution />

> #### *Form* (dua variabel) {/*form-two-variables*/}

Reviewer 3: ```suggestion
#### Form (dua variabel) {/*form-two-variables*/}
```

---

> ### Memperbarui *state* berdasarkan *state* sebelumnya {/*updating-state-based-on-the-previous-state*/}

Reviewer 3: ```suggestion
### Memperbarui state berdasarkan state sebelumnya {/*updating-state-based-on-the-previous-state*/}
```

---

> ### Memperbarui objek dan senarai di dalam *state*. {/*updating-objects-and-arrays-in-state*/}

Reviewer 3: ```suggestion
### Memperbarui objek dan senarai di dalam state {/*updating-objects-and-arrays-in-state*/}
```

});
```

Baca [memperbarui objek pada *state*](/learn/updating-objects-in-state) dan [memperbarui senarai pada *state](/learn/updating-arrays-in-state) untuk belajar lebih lanjut.

<Recipes titleText="Examples of objects and arrays in state" titleId="examples-objects">

> #### *Form* (objek) {/*form-object*/}

Reviewer 3: ```suggestion
#### Form (objek) {/*form-object*/}
```

<Solution />

> #### *Form* (objek bersarang) {/*form-nested-object*/}

Reviewer 3: ```suggestion
#### Form (objek bersarang) {/*form-nested-object*/}
```

---

> ### Menghindari membuat ulang *state* awal {/*avoiding-recreating-the-initial-state*/}

Reviewer 3: ```suggestion
### Menghindari membuat ulang state awal {/*avoiding-recreating-the-initial-state*/}
```

<Solution />

> #### Mengoper *state* awal secara langsung {/*passing-the-initial-state-directly*/}

Reviewer 3: ```suggestion
#### Mengoper state awal secara langsung {/*passing-the-initial-state-directly*/}
```

---

> ### Mereset *state* menggunakan kunci (*key*) {/*resetting-state-with-a-key*/}

Reviewer 3: ```suggestion
### Menyetel ulang state menggunakan kunci (key) {/*resetting-state-with-a-key*/}
```

reset -> setel ulang

---

> ### Menyimpan informasi dari *render* sebelumnya {/*storing-information-from-previous-renders*/}

Reviewer 3: ```suggestion
### Menyimpan informasi dari render sebelumnya {/*storing-information-from-previous-renders*/}
```

Pola ini bisa sulit dipahami dan biasanya lebih baik untuk dihindari. Namun, ini lebih baik daripada memperbarui *state* dalam efek. Ketika Anda memanggil fungsi `set` selama *render*, React akan me-*render* ulang komponen tersebut sesaat setelah komponen Anda keluar dengan pernyataan `return`, dan sebelum me-*render* anak-anak. Dengan begitu, anak-anak tidak perlu me-*render* dua kali. Sisa fungsi komponen Anda tetap akan dieksekusi (dan hasilnya akan dibuang). Jika kondisi Anda berada di bawah semua panggilan Hook, Anda dapat menambahkan `return` lebih awal untuk memulai ulang *render* sebelumnya.

---

> ## Pemecahan Masalah {/*troubleshooting*/}

Reviewer 3: ```suggestion
## Pemecahan masalah {/*troubleshooting*/}
```

Use sentence case for heading, as per global site styleguide.

> ### Saya sudah memperbarui *state*, tetapi saat saya mencetak log nilainya masih yang lama {/*ive-updated-the-state-but-logging-gives-me-the-old-value*/}

Reviewer 3: ```suggestion
### Saya sudah memperbarui state, tetapi saat saya mencetak log nilainya masih yang lama {/*ive-updated-the-state-but-logging-gives-me-the-old-value*/}
```

---

> ### Saya telah memperbarui *state*, tetapi tidak ada perbaruan pada layar {/*ive-updated-the-state-but-the-screen-doesnt-update*/}

Reviewer 3: ```suggestion
### Saya telah memperbarui state, tetapi tidak ada perbaruan pada layar {/*ive-updated-the-state-but-the-screen-doesnt-update*/}
```

---

> ### Saya mendapatkan galat: *"Too many re-renders"* {/*im-getting-an-error-too-many-re-renders*/}

Reviewer 3: ```suggestion
### Saya mendapatkan galat: "Too many re-renders" {/*im-getting-an-error-too-many-re-renders*/}
```

return <button onClick={(e) => handleClick(e)}>Click me</button>
```

Jika Anda tidak dapat menemukan penyebab kesalahan ini, klik pada panah di sebelah galat pada konsol dan lihat melalui tumpukan JavaScript untuk menemukan panggilan fungsi `set` yang bertanggung jawab atas kesalahan tersebut.

---

### Fungsi inisialisasi atau pembaruan saya berjalan dua kali {/*my-initializer-or-updater-function-runs-twice*/}

> Pada [*Strict Mod*e*](/reference/react/StrictMode), React akan memanggil beberapa fungsi Anda sebanyak dua kali alih-alih hanya satu kali:

Reviewer 3: ```suggestion
Pada [*Strict Mode*](/reference/react/StrictMode), React akan memanggil beberapa fungsi Anda sebanyak dua kali alih-alih hanya satu kali:
```

});
```

Apabila fungsi pembaruan ini murni, memanggilnya beberapa kali tidak membuat perbedaan perilaku. Inilah mengapa React memanggilnya dua kali untuk membantu Anda menemukan kesalahan. **Hanya fungsi komponen, inisialisasi, dan pembaruan yang perlu bersifat murni.** *Event handler* tidak perlu murni, sehingga React tidak akan pernah memanggil *event handler* Anda dua kali.

Baca [menjaga komponen murni](/learn/keeping-components-pure) untuk belajar lebih lanjut.

---

> ### Saya mencoba mengatur *state* ke sebuah fungsi, namun malah dipanggil {/*im-trying-to-set-state-to-a-function-but-it-gets-called-instead*/}

Reviewer 3: ```suggestion
### Saya mencoba mengatur state ke sebuah fungsi, namun malah dipanggil {/*im-trying-to-set-state-to-a-function-but-it-gets-called-instead*/}
```
